### PR TITLE
Open Liberty with Scala

### PR DESCRIPTION
**Pull request by krzsam**
Blog post to show an example Scala application implementing microservices in Open Liberty

### CODE REVIEW

File: drafts/2019-10-15-openliberty-with-scala.adoc [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+date:   2018-10-15 20:00:00 -0000
+categories: blog
+author_picture: https://avatars1.githubusercontent.com/u/50119017
+author_github: https://github.com/krzsam
+seo-title: Calculating Mandelbrot Set using OpenLiberty microservices implemented in Scala - OpenLiberty.io
+seo-description: An example application to calculate a Mandelbrot Set using Microprofile/Open Liberty microservices implemented in Scala. The goal is to demonstrate technical possibility of using OpenLiberty with Scala and possibility of deploying such solution as Kubernetes services.
+blog_description: "An example application to calculate a Mandelbrot Set using Microprofile/Open Liberty microservices implemented in Scala. The goal is to demonstrate technical possibility of using OpenLiberty with Scala and possibility of deploying such solution as Kubernetes services."
+---
+
+= Using Open Liberty in a Scala based application
```
Comment (lauracowen) [inline drafts/2019-10-15-openliberty-with-scala.adoc]: "Scala-based application" (hyphen)

File: drafts/2019-10-15-openliberty-with-scala.adoc [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+Krzysztof Samplawski <https://github.com/krzsam>
+
+This blog shows how Open Liberty microservices can be implemented using Scala language and deployed onto Kubernetes cluster.
```
Comment (lauracowen) [inline drafts/2019-10-15-openliberty-with-scala.adoc]: "on to a Kubernetes cluster."

File: drafts/2019-10-15-openliberty-with-scala.adoc [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+The main focus of the project was to explore technical feasibility of integration Open Liberty and Scala.
```
Comment (lauracowen) [inline drafts/2019-10-15-openliberty-with-scala.adoc]: "...explore the technical feasibility of integration between Open Liberty..."

File: drafts/2019-10-15-openliberty-with-scala.adoc [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+== Docker image creation
+The base Docker image for Open Liberty is already available on Docker Hub. Available documentation on https://openliberty.io/[openliberty.io] provides
+information how to use this Docker image when using Java and Maven toolset to create application image, and
```
Comment (lauracowen) [inline drafts/2019-10-15-openliberty-with-scala.adoc]: "to create an application image"

Comment (lauracowen) [inline drafts/2019-10-15-openliberty-with-scala.adoc]: "using the a Java and Maven toolset"

File: drafts/2019-10-15-openliberty-with-scala.adoc [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+the challenge was to analyze and replicate the same behaviour of creating an image but using Scala and sbt toolset, with some additional
```
Comment (lauracowen) [inline drafts/2019-10-15-openliberty-with-scala.adoc]: "using a Scala and sbt toolset"

File: drafts/2019-10-15-openliberty-with-scala.adoc [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+sbt plugins as required.
+An existing example application created by Maven was analyzed to investigate an internal structure of an application WAR file, which was
+then used to correctly parametrize sbt-docker plugin.
```
Comment (lauracowen) [inline drafts/2019-10-15-openliberty-with-scala.adoc]: "parameterize the sbt-docker plugin."

File: drafts/2019-10-15-openliberty-with-scala.adoc [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+Details on creating and testing Docker image can be found here: https://github.com/krzsam/scala-openliberty-mandelbrot/blob/master/README-Docker.md[README-Docker.md].
```
Comment (lauracowen) [inline drafts/2019-10-15-openliberty-with-scala.adoc]: "testing Docker images can be found"

File: drafts/2019-10-15-openliberty-with-scala.adoc [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+
+== Running on Kubernetes
+The application Docker image was deployed on Kubernetes cluster and all required deployments and services were created.
+The services worked correctly which showed no interoperability issues between Open Liberty and classes as compiled by Scala,
+all annotations defined on classes and methods were correctly retained in runtime and then correctly picked up and used by Open Liberty.
```
Comment (lauracowen) [inline drafts/2019-10-15-openliberty-with-scala.adoc]: "...as compiled by Scala. All notations defined on classes..."

File: drafts/2019-10-15-openliberty-with-scala.adoc [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+Details can be found here https://github.com/krzsam/scala-openliberty-mandelbrot/blob/master/README-k8s.md[README-k8s.md].
```
Comment (lauracowen) [inline drafts/2019-10-15-openliberty-with-scala.adoc]: "Details can be found in the https://github.com/krzsam/scala-openliberty-mandelbrot/blob/master/README-k8s.md[project README]." (if you agree?)

File: drafts/2019-10-15-openliberty-with-scala.adoc [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+
+== Notes on Performance
```
Comment (lauracowen) [inline drafts/2019-10-15-openliberty-with-scala.adoc]: "Notes on performance"

File: drafts/2019-10-15-openliberty-with-scala.adoc [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+It may be worth mentioning that the application is ready ro server requests in about 60s mark (as measured on the servers used),
```
Comment (lauracowen) [inline drafts/2019-10-15-openliberty-with-scala.adoc]: "ready for server requests in about 60 seconds (as measured on the servers used)," (is that the correct interpretation? I think there's a typo in your original.

File: drafts/2019-10-15-openliberty-with-scala.adoc [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+so this factor may need to be taken into account in use cases when it can be important.
+
+== Conclusion
+This example project showed that it is practically possible to use Open Liberty based microservices from a Scala application,
+and the ease of use and implementation is the same as if used in Java based applications.
```
Comment (lauracowen) [inline drafts/2019-10-15-openliberty-with-scala.adoc]: "is the same as with Java-based applications." (if you agree?)

File: drafts/2019-10-15-openliberty-with-scala.adoc [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+The project can be found here https://github.com/krzsam/scala-openliberty-mandelbrot[GitHub: scala-openliberty-mandelbrot]
```
Comment (lauracowen) [inline drafts/2019-10-15-openliberty-with-scala.adoc]: "The project can be found at https:...."

File: drafts/2019-10-15-openliberty-with-scala.adoc [hunk 14]
```diff
@@ -0,0 +1,41 @@
+---
+layout: post
+title: "Calculating Mandelbrot Set using OpenLiberty microservices implemented in Scala"
+date:   2018-10-15 20:00:00 -0000
+categories: blog
+author_picture: https://avatars1.githubusercontent.com/u/50119017
+author_github: https://github.com/krzsam
+seo-title: Calculating Mandelbrot Set using OpenLiberty microservices implemented in Scala - OpenLiberty.io
+seo-description: An example application to calculate a Mandelbrot Set using Microprofile/Open Liberty microservices implemented in Scala. The goal is to demonstrate technical possibility of using OpenLiberty with Scala and possibility of deploying such solution as Kubernetes services.
```
[the remark this answers is not in the document]
Comment (krzsam) [reply drafts/2019-10-15-openliberty-with-scala.adoc]: corrected line 9 and 10

File: drafts/2019-10-15-openliberty-with-scala.adoc (hunk 1 above)
Comment (krzsam) [reply drafts/2019-10-15-openliberty-with-scala.adoc]: corrected

File: drafts/2019-10-15-openliberty-with-scala.adoc (hunk 2 above)
Comment (krzsam) [reply drafts/2019-10-15-openliberty-with-scala.adoc]: corrected

File: drafts/2019-10-15-openliberty-with-scala.adoc (hunk 3 above)
Comment (krzsam) [reply drafts/2019-10-15-openliberty-with-scala.adoc]: corrected

File: drafts/2019-10-15-openliberty-with-scala.adoc (hunk 4 above)
Comment (krzsam) [reply drafts/2019-10-15-openliberty-with-scala.adoc]: corrected

Comment (krzsam) [reply drafts/2019-10-15-openliberty-with-scala.adoc]: corrected

File: drafts/2019-10-15-openliberty-with-scala.adoc (hunk 5 above)
Comment (krzsam) [reply drafts/2019-10-15-openliberty-with-scala.adoc]: corrected

File: drafts/2019-10-15-openliberty-with-scala.adoc (hunk 6 above)
Comment (krzsam) [reply drafts/2019-10-15-openliberty-with-scala.adoc]: corrected

File: drafts/2019-10-15-openliberty-with-scala.adoc (hunk 7 above)
Comment (krzsam) [reply drafts/2019-10-15-openliberty-with-scala.adoc]: corrected, phrased similar 3 references in same consistent way

File: drafts/2019-10-15-openliberty-with-scala.adoc (hunk 8 above)
Comment (krzsam) [reply drafts/2019-10-15-openliberty-with-scala.adoc]: corrected

File: drafts/2019-10-15-openliberty-with-scala.adoc (hunk 9 above)
Comment (krzsam) [reply drafts/2019-10-15-openliberty-with-scala.adoc]: Rephrased the link text in similar way across 2 references in lines 26 and 32, hope it looks better now

File: drafts/2019-10-15-openliberty-with-scala.adoc (hunk 10 above)
Comment (krzsam) [reply drafts/2019-10-15-openliberty-with-scala.adoc]: corrected

File: drafts/2019-10-15-openliberty-with-scala.adoc (hunk 11 above)
Comment (krzsam) [reply drafts/2019-10-15-openliberty-with-scala.adoc]: corrected the typo so the sentence should make more sense now

File: drafts/2019-10-15-openliberty-with-scala.adoc (hunk 12 above)
Comment (krzsam) [reply drafts/2019-10-15-openliberty-with-scala.adoc]: corrected

File: drafts/2019-10-15-openliberty-with-scala.adoc (hunk 13 above)
Comment (krzsam) [reply drafts/2019-10-15-openliberty-with-scala.adoc]: corrected

**Comment by krzsam** [comment]
Thank you for the review - all the places have been corrected (hope I did not miss anything) so it should look better now.

**Comment by krzsam** [comment]
Awesome ! No problem, that was real fun to try.

**Pull request merged**